### PR TITLE
Add jwt authenticator to login module

### PR DESCRIPTION
**Pull request by SabraO**

### CODE REVIEW

File: modules/jaggeryapps/appmgt/src/modules/user/login/login.jag [hunk 1]
```diff
@@ -130,4 +130,149 @@ function logout(){
 
 }
 
+function jwtisUserAuthenticated(authorizationHeader) {
+    log.debug("Initializing jwt login module");
```
Comment (amalkasubasinghe) [inline modules/jaggeryapps/appmgt/src/modules/user/login/login.jag]: wrap with log.isDebugEnabled()

File: modules/jaggeryapps/appmgt/src/modules/user/login/login.jag [hunk 2]
```diff
@@ -130,4 +130,149 @@ function logout(){
 
 }
 
+function jwtisUserAuthenticated(authorizationHeader) {
```
Comment (amalkasubasinghe) [inline modules/jaggeryapps/appmgt/src/modules/user/login/login.jag]: jwtisUserAuthenticated -> 'is' should be 'Is'

File: modules/jaggeryapps/appmgt/src/modules/user/login/login.jag [hunk 3]
```diff
@@ -130,4 +130,149 @@ function logout(){
...
+    log.debug("Initializing jwt login module");
+    var startTime_module = new Date().getTime();
+
+    var tenantManager = multitenancy.getTenantManager();
+    var KeyStoreManager = Packages.org.wso2.carbon.core.util.KeyStoreManager;
```
Comment (amalkasubasinghe) [inline modules/jaggeryapps/appmgt/src/modules/user/login/login.jag]: variable names should starts with simple letter

File: modules/jaggeryapps/appmgt/src/modules/user/login/login.jag (hunk 1 above)
Comment (SabraO) [inline modules/jaggeryapps/appmgt/src/modules/user/login/login.jag]: fixed with 89e797b

File: modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/ApplicationManager.java [hunk 4]
```diff
@@ -1428,4 +1428,26 @@ public static void updateVersionExposureLevel(String exposureLevel, String versi
...
     }
+
+    /**
+     * Method to get custom domain details for tenant
+     *
+     * @param tenantId tenant id
+     * @return list of custom domain details for all applications of tenant
+     * @throws AppCloudException
+     */
+    public static Application[] getCustomDomainDetailsForTenant(int tenantId) throws AppCloudException {
```
Comment (SabraO) [inline modules/components/org.wso2.appcloud.core/src/main/java/org/wso2/appcloud/core/ApplicationManager.java]: fixed at bd16475e112e95591327a8c07064e6a2f4e622d0

File: modules/jaggeryapps/appmgt/src/modules/user/login/login.jag [hunk 5]
```diff
@@ -130,4 +130,142 @@ function logout(){
...
+    if (splitValues == null || splitValues.length == 0) {
+        if (log.isDebugEnabled()) {
+            log.debug("Authorization Type is not defined. Hence returning null");
+        }
+        return null;
+    }
+    return splitValues[0].trim();
+}
+
+function decodeAuthorizationHeader(authorizationHeader) {
```
Comment (SabraO) [inline modules/jaggeryapps/appmgt/src/modules/user/login/login.jag]: Add method comments and null check.

File: modules/jaggeryapps/appmgt/src/modules/user/login/login.jag [hunk 6]
```diff
@@ -130,4 +130,142 @@ function logout(){
...
+        msg = "Error authenticating the user for authorization header: " + authorizationHeader;
+        log.error(msg);
+        log.error(e);
+    } finally {
+        context.endTenantFlow();
+    }
+}
+
+function getAuthType(authorizationHeader) {
+    var splitValues = null;
```
Comment (SabraO) [inline modules/jaggeryapps/appmgt/src/modules/user/login/login.jag]: Improve code logic

File: modules/jaggeryapps/appmgt/src/modules/user/login/login.jag (hunk 5 above)
Comment (SabraO) [inline modules/jaggeryapps/appmgt/src/modules/user/login/login.jag]: fixed @ d4d2b3d31d7ac2fa730bc10c4f7388b361da3dbf

File: modules/jaggeryapps/appmgt/src/modules/user/login/login.jag (hunk 6 above)
Comment (SabraO) [inline modules/jaggeryapps/appmgt/src/modules/user/login/login.jag]: fixed @ d4d2b3d31d7ac2fa730bc10c4f7388b361da3dbf

**Pull request merged**